src/main/java/seedu/address/model/task/TaskList.java

```diff
@@ -0,0 +1,93 @@
+package seedu.address.model.task;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.task.exceptions.TaskNotFoundException;
+
+/**
+ * A list of Tasks that enforces uniqueness between its elements and does not allow nulls.
+ * (inspired from UniqueEmployeeList.java)
+ */
+public class TaskList implements Iterable<Task> {
```

Comment:
Perhaps this class should be renamed to `UniqueTaskList` to be in line with the other unique lists?

Reply:
It's not unique, that's why i didn't name it as unique. We should further discuss the implementation in future PRs